tests/test_recipes.py

```diff
@@ -65,11 +112,50 @@ def test_prune_recipe(recipe_fixture, execute_recipe, nkeep):
 @pytest.mark.parametrize("cache_inputs", [True, False])
 @pytest.mark.parametrize("copy_input_to_local_file", [True, False])
 def test_recipe_caching_copying(
-    netCDFtoZarr_sequential_recipe, execute_recipe, cache_inputs, copy_input_to_local_file
+    netCDFtoZarr_recipe, execute_recipe, cache_inputs, copy_input_to_local_file
 ):
-    """The basic recipe test. Use this as a template for other tests."""
+    """Test that caching and copying to local file work."""
+
+    RecipeClass, file_pattern, kwargs, ds_expected, target = netCDFtoZarr_recipe
+
+    if not cache_inputs:
+        kwargs.pop("input_cache")  # make sure recipe doesn't require input_cache
+    rec = RecipeClass(
+        file_pattern,
+        **kwargs,
+        cache_inputs=cache_inputs,
+        copy_input_to_local_file=copy_input_to_local_file
+    )
+    execute_recipe(rec)
+    ds_actual = xr.open_zarr(target.get_mapper()).load()
+    xr.testing.assert_identical(ds_actual, ds_expected)
+
+
+@pytest.mark.parametrize("cache_inputs", [True, False])
+@pytest.mark.parametrize("copy_input_to_local_file", [True, False])
+def test_recipe_http_caching_copying(
+    netCDFtoZarr_http_recipe, execute_recipe, cache_inputs, copy_input_to_local_file
+):
+    """Test that caching and copying from http to local file work."""
+
+    RecipeClass, file_pattern, kwargs, ds_expected, target = netCDFtoZarr_http_recipe
+
+    if len(file_pattern.merge_dims) != 0:
+        pytest.skip(
+            "This test ensures auth kwargs are correctly passed to the recipe class."
+            "`netcdf_http_file_pattern` fixture currently does not return a properly"
+            " formatted `path_format` for recipes with a `MergeDim.`"
+        )
+
+    first_url = list(file_pattern.items())[0][1]
+    r = requests.get(first_url)
+    if r.status_code == 401:
+        fsspec_open_kwargs = dict(auth=aiohttp.BasicAuth("foo", "bar"))
+        kwargs.update({"fsspec_open_kwargs": fsspec_open_kwargs})
+    elif r.status_code == 400:
+        query_string_secrets = {"foo": "foo", "bar": "bar"}
+        kwargs.update({"query_string_secrets": query_string_secrets})
```

Comment:
This seems like a very roundabout way to figure out if we need to use secrets. Perhaps we should be embedding the secrets _inside_ the FilePattern object.